3.2.2+ update

r2dliu opened the pull request.

JosJuice committed:
ControllerInterface/Android: Rip out ButtonManager

ButtonManager is very different from how a normal input backend works,
and is making it hard for us to improve controller support on Android.
The following commits will add a new input backend in its place.

JosJuice committed:
Android: Stop extracting pre-defined controller INIs

JosJuice committed:
Android: Remove copyAsset's overwrite parameter

Because of the previous commit, this is no longer needed.

JosJuice committed:
ControllerInterface/Android: Implement device population

JosJuice committed:
ControllerInterface/Android: Handle input events

Android doesn't let us poll inputs whenever we want. Instead, we
listen to input events (activities will have to forward them to the
input backend), and store the received values in atomic variables
in the Input classes. This is similar in concept to how ButtonManager
worked, but without its homegrown second input mapping system.

JosJuice committed:
Android: Add new input mapping implementation

JosJuice committed:
Android: Remove old input mapping implementation

This old code is of no use now that ButtonManager is gone.

JosJuice committed:
ControllerEmu: Add default mappings for Android

Unlike PCs, Android doesn't really have any input method (not counting
touch) that can reasonably be expected to exist on most devices.
Because of this, I don't think shipping with a default mapping for the
buttons and sticks of GameCube controllers and Wii Remotes makes sense.
I would however like to ship default mappings for a few things:

1. Mapping the Wii Remote's accelerometer and gyroscope to the device's
   accelerometer and gyroscope. This functionality is useful mainly
   for people who use the touchscreen, but can also be useful when
   using a clip-on controller. The disadvantage of having this mapped
   by default is that games disable pointer input if the accelerometer
   reports that the Wii Remote is pointed at the ceiling.

2. Mapping GC keyboards for use with a physical keyboard, like on PC.
   After all, there's no other way of mapping them that makes sense.

3. Mapping rumble to the device's vibrator.

Aside from the GC keyboards, this approach is effectively the same as
what we were doing before the input overhaul.

JosJuice committed:
ControllerInterface/Android: Return whether input was handled

When Android presents an input event to an app, it wants the app to
return true or false depending on whether the app handled the event or
not. If the event wasn't handled by the app, it will be passed on to
the system, which may decide to take an action depending on what kind
of input event it is. For instance, if a B button press is passed on to
the system, it will be turned into a Back press. But if an R1 press is
passed on to the system, nothing in particular happens.

It's important that we get this return value right in Dolphin. For
instance, the user generally wouldn't want a B button press to open
the EmulationActivity menu, so B button presses usually shouldn't be
passed on to the system - but volume button presses usually should be
passed on to the system, since it would be hard to adjust the volume
otherwise. What ButtonManager did was to pass on input events that are
for a button which the user has not mapped, which I think makes sense.
But exactly how to implement that is more complicated in the new input
backend than in ButtonManager, because now we have a separation between
the input backend and the code that keeps track of the user's mappings.

What I'm going with in this commit is to treat an input as mapped if
it has been polled recently. In part I chose this because it seemed
like a simple way of implementing it that wouldn't cause too many
layering violations, but it also has two useful side effects:

1. If a controller is not being polled (e.g. GameCube controllers in
   Wii games that don't use them), its mappings will not be considered.
2. Once sensor input is implemented in the Android input backend,
   we will be able to use this "polled recently" tracking to power down
   the sensors at times when the game is using a Wii Remote reporting
   mode that doesn't include motion data. (Assuming that the sensor
   inputs only are mapped to Wii Remote motion controls, that is.)

JosJuice committed:
ControllerInterface/Android: Implement hotplug

JosJuice committed:
Android: Add NumericSetting support

JosJuice committed:
Android: Add extension selection

JosJuice committed:
Android: Split Wii Remote settings into submenus

Too much stuff on one screen otherwise. This split matches DolphinQt.

JosJuice committed:
GCPadEmu/WiimoteEmu: Reorder control groups

This way, Android (which will show groups in the order they're defined)
will show groups in a more logical order similar to DolphinQt.

The main thing that was annoying me was how early Rumble was for
Wii Remotes. Some of the other changes I'm making in this commit,
like the order of Shake/Tilt/Swing, are more arbitrary and were
made for consistency with DolphinQt. But there are also places
where I didn't go all the way with matching DolphinQt. Most notably,
DolphinQt puts sticks before buttons, but I don't see any reason
to do that for Android.

JosJuice committed:
Android: Implement enabling/disabling control groups

All this code for just a single checkbox... Ah well, it has to be done

JosJuice committed:
Android: Allow changing controller settings during emulation

NikhilNarayana commented:
To do:
https://github.com/project-slippi/Ishiiruka/commit/f84543e623c14e80f58b73b8d3c380328d61b6ac
https://github.com/project-slippi/Ishiiruka/commit/d35ba2bd04730faa516cfb89910ce63da77a9474

JordanLozinski commented:
> TODO:
> 
> ...
> * [project-slippi/Ishiiruka@d35ba2b](https://github.com/project-slippi/Ishiiruka/commit/d35ba2bd04730faa516cfb89910ce63da77a9474)
> ...


I'm working on this one.

JosJuice committed:
Android: Remove the in-emulation motion control options

You can set this in the normal controller settings now.

JosJuice committed:
Android: Remove the in-emulation IR sensitivity options

This too can be set in the normal controller settings now.

JosJuice committed:
Android: Change how the overlay controller setting works

Up until now, there have been two settings on Android that stored the
selected Wii Remote extension: the normal one that's also used on PC,
and a SharedPreferences one that's used by the overlay controls to
determine what controls to show. It is possible for these two to end up
out of sync, and my input changes have made that more likely to happen.

To fix this, let's rework how the overlay controller setting works.
We don't want it to encode the currently selected Wii Remote extension.
However, we can't simply get rid of the setting, because for some Wii
games we need the ability to switch between a GameCube controller and a
Wii Remote. What this commit does is give the user the option to select
any of the 4 GameCube controllers and any of the 4 Wii Remotes. (Before,
controllers 2-4 weren't available in the overlay.) Could be useful for
things like the Psycho Mantis fight in Metal Gear Solid. I'm also
switching from SharedPreferences to Dolphin.ini while I'm at it.

JosJuice committed:
Android: Change how the overlay controller setting works, part 2

The previous commit wasn't enough for getting inputs to work for
controllers 2-4.

Minty-Meeo committed:
[[unlikely]] ASSERT

and other ASSERT usage changes

JosJuice committed:
Android: Add input device selection

JosJuice committed:
Android: Add Default and Clear actions for controllers

JosJuice committed:
Android: Show warning when using old pre-defined controller INIs

Co-authored-by: Charles Lombardo <clombardo169@gmail.com>

JosJuice committed:
ControllerInterface/Android: Implement sensor input

JosJuice committed:
Android: Remove old motion input implementation

JosJuice committed:
ControllerInterface/Android: Implement sensor input for InputDevices

This functionality was added in Android 12 to let apps get motion data
for gamepads.

JosJuice committed:
ControllerInterface/Android: Automatically suspend sensors

This is a battery-saving measure. Whether a sensor should be suspended
is determined in the same way as whether key events and motion events
should be handled by the OS rather than consumed by Dolphin.

JosJuice committed:
Android: Move Sideways Wii Remote setting one submenu up

A slight departure from DolphinQt, but I think it makes sense, because
it will make the important Sideways Wii Remote setting easier to find.

JosJuice committed:
Android: Add "more settings" button to overlay control setting dialog

Some people might wonder where the ability to select an extension
and the Sideways Wii Remote went. This button will take them to the
general settings, which is where those settings now live.

At some point in the future, we should probably move everything to the
general settings. But this pull request is already big enough as it is!

JosJuice committed:
Android: Remove old rumble implementation

JosJuice committed:
ControllerInterface/Android: Implement rumble

JosJuice committed:
DolphinQt: Add MotionPlus support to TAS input

Will manually controlling both an accelerometer and a gyroscope at the
same time be reasonable to do? No idea. Was this easy to implement
thanks to the input override system? Yes.

Fixes https://bugs.dolphin-emu.org/issues/12443.

JosJuice committed:
Android: Add "Create mappings for other devices"

JosJuice committed:
Android: Prevent showing mapping dialog with no device selected

JosJuice committed:
Android: Add input profile management

Co-authored-by: Charles Lombardo <clombardo169@gmail.com>

JosJuice committed:
Android: Add selecting input profile per game

JosJuice committed:
Config: Simplify IsSettingSaveable's handling of Android section

The Android-specific controller mapping system is now gone,
so IsSettingSaveable can be greatly simplified.

JosJuice committed:
Android: Add the advanced input mapping dialog

It's missing a lot of features from the PC version for now, like
buttons for inserting functions and the ability to see what the
expression evaluates to. I mostly just wanted to get something in
place so you can set up rumble.

Co-authored-by: Charles Lombardo <clombardo169@gmail.com>

JosJuice committed:
Jit: Don't use a second stack

This second stack leads to JNI problems on Android, because ART fetches
the address and size of the original stack using pthread functions
(see GetThreadStack in art/runtime/thread.cc), and (presumably) treats
stack addresses outside of the original stack as invalid. (What I don't
understand is why some JNI operations on the CPU thread work fine
despite this but others don't.)

Instead of creating a second stack, let's borrow the approach ART uses:
Use pthread functions to find out the stack's address and size, then
install guard pages at an appropriate location. This lets us get rid
of a workaround we had in the MsgAlert function.

Because we're no longer choosing the stack size ourselves, I've made some
tweaks to where the put the guard pages. Previously we had a stack of
2 MiB and a safe zone of 512 KiB. We now accept stacks as small as 512 KiB
(used on macOS) and use a safe zone of 256 KiB. I feel like this should
be fine, but haven't done much testing beyond "it seems to work".

By the way, on Windows it was already the case that we didn't create
a second stack... But there was a bug in the implementation!
The code for protecting the stack has to run on the CPU thread, since
it's the CPU thread's stack we want to protect, but it was actually
running on EmuThread. This commit fixes that, since now this bug
matters on other operating systems too.

t895 committed:
Android: Convert SystemUpdateProgressBarDialogFragment to Kotlin

t895 committed:
Android: Convert SystemUpdateViewModel to Kotlin

t895 committed:
Android: Convert WiiUtils to Kotlin

t895 committed:
Android: Convert SystemUpdateResultFragment to Kotlin

t895 committed:
Android: Convert SystemMenuNotInstalledDialogFragment to Kotlin

t895 committed:
Android: Convert OnlineUpdateRegionSelectDialogFragment to Kotlin

t895 committed:
Android: Use Kotlin extensions for lifecycle components

t895 committed:
Android: Remove orientation lock on system update dialog

t895 committed:
Android: Remove hardcoded strings from system update dialogs

t895 committed:
Android: Switch to indeterminate progress bar on system update cancel

t895 committed:
Android: Use coroutine for system updates

t895 committed:
Android: Use DialogFragments to direct UserData actions

K0bin committed:
Android: Implement DocumentProvider

This allows users to access the Dolphin user directory.

AdmiralCurtiss committed:
DiscIO: Add support for CleanRip-style split ISOs.

Pokechu22 committed:
Software: Fix crash on startup when using "Compile Shaders Before Starting"

When that setting is enabled, m_xfb_entry is initially not present (during the phase where a shader compilation progress bar would be shown). The main path checks for m_xfb_entry, but the software renderer fallback path didn't.

Fixes another aspect of https://bugs.dolphin-emu.org/issues/13172.

Pokechu22 committed:
Software: Implement GetSurfaceInfo()

Before, it used a fallback where it returned a default object, where the width and height were set to 0. Presenter::Initialize() used GetSurfaceInfo to set the backbuffer size, then used that size when initializing the on-screen UI (even for the software renderer, where the on-screen UI isn't currently present), which meant that ImGui got a window size of 0 and thus resulted in a failed assertion.

Although BindBackbuffer checks for size changes, it doesn't help because ImGui has already been initialized, and the size hasn't actually changed since initialization occured.

Fixes one aspect of https://bugs.dolphin-emu.org/issues/13172.

Pokechu22 committed:
Software: Fix "Auto-Adjust Window Size"

This also needs to be handled on the software renderer path.

AdmiralCurtiss committed:
Common/FatFsUtil: Add callback for cancelling SD card conversion.

AdmiralCurtiss committed:
Qt/WiiPane: Add progress window for SD card conversion.

AdmiralCurtiss committed:
Common/FatFsUtil: Close temp file before deleting it on conversion failure.

JosJuice committed:
Android: Hide controller mappings button when controller type is None

Also removed the make_sure_continuous_scan_enabled message.
It doesn't make sense with the new UX.

JosJuice committed:
Jit: Move stack handling code to JitBase

Deduplication between Jit64 and JitArm64.

iwubcode committed:
VideoCommon: add class to load custom texture data

iwubcode committed:
VideoCommon: remove HiResTexture DDS loading, update hirestexture logic to use custom texture data

t895 committed:
Android: Convert ThreadUtil to Kotlin

t895 committed:
Merge pull request #11515 from t895/user-data-kotlin

Android: Rewrite User Data Activity in Kotlin

t895 committed:
Android: Initialize TaskViewModel earlier in User Data Activity

JosJuice committed:
Merge pull request #11613 from t895/import-fix

Android: Initialize TaskViewModel earlier in User Data Activity

t895 committed:
Android: Convert ARCheat to Kotlin

t895 committed:
Android: Convert PatchCheat to Kotlin

t895 committed:
Android: Convert GraphicsModGroup to Kotlin

t895 committed:
Android: Convert GraphicsMod to Kotlin

t895 committed:
Android: Convert GeckoCheat to Kotlin

t895 committed:
Android: Convert AbstractCheat to Kotlin

t895 committed:
Android: Convert Cheat to Kotlin

t895 committed:
Android: Convert CheatsViewModel to Kotlin

t895 committed:
Android: Convert ReadOnlyCheat to Kotlin

t895 committed:
Android: Convert ActionViewHolder to Kotlin

t895 committed:
Android: Convert CheatDetailsFragment to Kotlin

t895 committed:
Android: Convert CheatItem to Kotlin

t895 committed:
Android: Convert CheatItemViewHolder to Kotlin

t895 committed:
Android: Convert CheatListFragment to Kotlin

t895 committed:
Android: Convert CheatsActivity to Kotlin

t895 committed:
Android: Convert CheatsAdapter to Kotlin

t895 committed:
Android: Convert CheatsDisabledWarningFragment to Kotlin

t895 committed:
Android: Convert CheatViewHolder to Kotlin

t895 committed:
Android: Convert GraphicsModsDisabledWarningFragment to Kotlin

t895 committed:
Android: Convert HeaderViewHolder to Kotlin

t895 committed:
Android: Convert SettingDisabledWarningFragment to Kotlin

t895 committed:
Android: Convert TwoPaneOnBackPressedCallback to Kotlin

t895 committed:
Android: Convert RiivolutionPatches to Kotlin

t895 committed:
Android: Convert RiivolutionViewHolder to Kotlin

t895 committed:
Android: Convert RiivolutionItem to Kotlin

t895 committed:
Android: Convert RiivolutionBootActivity to Kotlin

t895 committed:
Android: Convert RiivolutionAdapter to Kotlin

t895 committed:
Android: Convert Skylander to Kotlin

t895 committed:
Android: Convert SkylanderPair to Kotlin

t895 committed:
Android: Convert SkylanderConfig to Kotlin

t895 committed:
Android: Convert SkylanderSlot to Kotlin

t895 committed:
Android: Convert SkylanderSlotAdapter to Kotlin

t895 committed:
Android: Convert ConvertFragment to Kotlin

t895 committed:
Android: Convert ConvertActivity to Kotlin

SMarioMan committed:
Avoid altering TCP sockets

Co-authored-by: sepalani <sepalani@hotmail.fr>

lioncash committed:
Merge pull request #11619 from t895/kotlin-convert

Android: Convert Convert Activity to Kotlin

lioncash committed:
Merge pull request #11616 from t895/kotlin-skylanders

Android: Convert Skylanders code to Kotlin

lioncash committed:
Merge pull request #11615 from t895/kotlin-riivolution

Android: Convert Riivolution Boot Activity to Kotlin

lioncash committed:
Merge pull request #11614 from t895/kotlin-cheats

Android: Convert Cheats Activity to Kotlin

lioncash committed:
Merge pull request #11612 from iwubcode/custom_texture_data

VideoCommon: move hirestexture data to reusable class called CustomTextureData

lioncash committed:
Merge pull request #11605 from JosJuice/android-mappings-button

Android: Add a button for accessing controller mappings

lioncash committed:
Merge pull request #11607 from AdmiralCurtiss/sd-progress

Qt/WiiPane: Add progress window for SD card conversion.

lioncash committed:
Merge pull request #11582 from Pokechu22/software-crash-on-startup

Software: Fix regressions from "Kill Renderer"

Pokechu22 committed:
Merge pull request #10908 from Minty-Meeo/unlikely-assert

[[unlikely]] ASSERT and minor ASSERT usage changes

JosJuice committed:
Merge pull request #11399 from JosJuice/jit-one-stack

Jit: Don't use a second stack

deReeperJosh committed:
Skylander Portal: Clearing and Reloading fix

JosJuice committed:
DolphinQt: Rework TAS input threading, part 1 (buttons)

This gets rid of a blocking operation, improving performance and fixing
https://bugs.dolphin-emu.org/issues/12893.

This also makes us no longer directly access the state of certain UI
elements from the CPU thread, which probably wasn't thread-safe but
doesn't seem to have caused any observable issues so far.

AdmiralCurtiss committed:
Qt/WatchWidget: Don't update if not paused.

phire committed:
Fix duplicated FPS

This is a small regression from KillRenderer, which caused duplicated
frames to be counted on the FPS counter when the "Skip Presenting
Duplicated Frames" option was disabled.

phire committed:
Merge pull request #11624 from phire/fix_duplicate_fps

Fix duplicated FPS

phire committed:
Don't discard ConfigChanged callback handle

phire committed:
HookableEvents: Add [[nodiscard]] to Register

phire committed:
Merge pull request #11625 from phire/eventhook_nodiscard

Fix KillRenderer config changed regression

CasualPokePlayer committed:
Update mGBA to 59ebf1c12dcbbe0141000cddccf667cd42e386a1

AdmiralCurtiss committed:
Merge pull request #11626 from CasualPokePlayer/mgba_update_clang-cl

Update mgba to 59ebf1c12dcbbe0141000cddccf667cd42e386a1

AdmiralCurtiss committed:
Update mGBA submodule to 89f8873df366a99dfdf40347f72e68763fb50bd9

Fixes regression that broke importing save files.

JosJuice committed:
Android: Disable input device sensor input due to crash

AdmiralCurtiss committed:
HW/HSP: Refactor to class, move to System.

AdmiralCurtiss committed:
HW/CPU: Refactor to class, move to System.

lioncash committed:
Merge pull request #11629 from AdmiralCurtiss/globals-hsp

HW/HSP: Refactor to class, move to System.

JosJuice committed:
DolphinQt: Rework TAS input threading, part 2 (analog inputs)

JosJuice committed:
DolphinQt: Remove unneeded out parameters in TASInputWindow

IIRC we needed this before the input override system was added.

AdmiralCurtiss committed:
Merge pull request #11623 from AdmiralCurtiss/watch-window-workaround

Qt/WatchWidget: Don't update if not paused.

AdmiralCurtiss committed:
Merge pull request #11627 from AdmiralCurtiss/mgba-savegame-fix

Update mGBA submodule to 89f8873df366a99dfdf40347f72e68763fb50bd9

lioncash committed:
Merge pull request #11621 from deReeperJosh/skylandersfilefix

Skylander Portal: Clearing and Reloading fix

AdmiralCurtiss committed:
Merge pull request #11568 from SMarioMan/socket-broadcast

Network/Socket: Enable broadcast permissions in socket requests

AdmiralCurtiss committed:
Core: Add System parameter to CPUThreadGuard.

AdmiralCurtiss committed:
Core/Boot: Pass around System.

lioncash committed:
Merge pull request #11494 from t895/orientation-jank

Android: Rewrite Wii system updates in Kotlin

K0bin committed:
Android: Only show divider in settings for headers

lioncash committed:
Merge pull request #11630 from AdmiralCurtiss/globals-cpu

HW/CPU: Refactor to class, move to System.

K0bin committed:
Android: Center list view headers

AdmiralCurtiss committed:
HW/EXI: Pass System to IEXIDevice.

AdmiralCurtiss committed:
HW/AudioInterface: Refactor to class.

lioncash committed:
Merge pull request #11632 from AdmiralCurtiss/pass-more-system

Pass System to Boot functions, CPUThreadGuard, IEXIDevice.

AdmiralCurtiss committed:
HW/DSP: Refactor to class.

t895 committed:
Merge pull request #11631 from K0bin/pref-divider

Android: Only show divider in settings for headers

AdmiralCurtiss committed:
Merge pull request #11622 from JosJuice/tas-input-nonblocking

DolphinQt: Rework TAS input threading

AdmiralCurtiss committed:
Merge pull request #11601 from Dentomologist/bluetooth_adapter_missing_message

Config: Restore Bluetooth adapter missing message in Controller Settings

AdmiralCurtiss committed:
Merge pull request #11557 from AdmiralCurtiss/split-file-reader

DiscIO: Add support for CleanRip-style split ISOs.

AdmiralCurtiss committed:
Merge pull request #11441 from JosJuice/jit64-lea-fixup

Jit64: Fix the offsetAddedToAddress correction

AdmiralCurtiss committed:
Merge pull request #11562 from kaiboisdumb/patch-1

Disable Immediate XFB in Cubivore

lioncash committed:
Merge pull request #11633 from AdmiralCurtiss/ai-class

HW/AudioInterface: Refactor to class.

AdmiralCurtiss committed:
HW/DVDInterface: Refactor to class.

lioncash committed:
Merge pull request #11634 from AdmiralCurtiss/dsp-class

HW/DSP: Refactor to class.

AdmiralCurtiss committed:
HW/DVDThread: Refactor to class.

AdmiralCurtiss committed:
HW/DVD: Rename DVDInterface namespace to DVD.

AdmiralCurtiss committed:
HW/DVD: Rename DVDInterfaceManager to DVDInterface.

AdmiralCurtiss committed:
HW/DVD: Merge DVDThread namespace into DVD.

AdmiralCurtiss committed:
HW/DVD: Rename DVDThreadManager to DVDThread.

AdmiralCurtiss committed:
HW/EXI: Refactor ExpansionInterface to class.

AdmiralCurtiss committed:
HW/MemoryInterface: Refactor to class.

AdmiralCurtiss committed:
HW/VideoInterface: Refactor to class.

lioncash committed:
Merge pull request #11637 from AdmiralCurtiss/dvd-class

HW/DVD: Refactor DVDInterface and DVDThread to classes.

lioncash committed:
Merge pull request #11638 from JosJuice/tas-input-motionplus

DolphinQt: Add MotionPlus support to TAS input

AdmiralCurtiss committed:
HW/VideoInterface: Savestate members individually.

lioncash committed:
Merge pull request #11639 from AdmiralCurtiss/exi-class

HW: Refactor ExpansionInterface and MemoryInterface to classes.

JosJuice committed:
Merge pull request #11385 from JosJuice/android-input-overhaul

Android input overhaul

JosJuice committed:
Merge pull request #11524 from K0bin/document-provider

Android: Implement DocumentProvider

shuffle2 committed:
winupdater: improve wording of UI::Error

lioncash committed:
Merge pull request #11640 from AdmiralCurtiss/vi-class

HW/VideoInterface: Refactor to class.

InfoManiac742 committed:
Create GSB.ini

lioncash committed:
Merge pull request #11642 from shuffle2/winupdater-fail

winupdater: improve wording of UI::Error

lioncash committed:
Merge pull request #11643 from InfoManiac742/master

Fix missing GameINI for Japanese version of Sonic Adventure 2: Battle

K0bin committed:
Android: Use $applicationId for DocmentProvider authority

Fixes side by side installation of debug, benchmark and release builds.

JosJuice committed:
Merge pull request #11645 from K0bin/document-provider-authority-fix

Android: Use $applicationId for DocmentProvider authority

NikhilNarayana committed:
Merge branch 'master' of https://github.com/dolphin-emu/dolphin into pull-upstream-2023-02-25

r2dliu committed:
fix all compile errors. no functionality yet (#77)

* fix all compile errors. can't link zlib

* fix oopsie

* nogui still has zlib error. need to unlink in public. dolphin-gui compiles and runs!

* fix cmakelists include

r2dliu committed:
most of ini stuff converted. Still need to build Qt UI. Slot B not yet defaulting to slippi for some reason

r2dliu committed:
playback work. File loads but replay freezes shortly after. stripped out a lot of ui and playback functionality to get compiling

NikhilNarayana committed:
Merge pull request #76 from r2dliu/pull-upstream-2023-02-25

NikhilNarayana committed:
ci: update runners

r2dliu committed:
pull in enormous ranked commit e7adf9f79ef2c3d60066bbb36d672da841560dab. add missing data/sys files from current head, f68f6f6cbf56b3e4e43f27475763f1d15b6d7c67

r2dliu committed:
Merge branch 'task/update' of https://github.com/r2dliu/dolphin into task/update

r2dliu committed:
fix compile

r2dliu committed:
even with d7f10754b27fd216bf2180dff15639e39ec3bdeb. add injection list files from head at f68f6f6cbf56b3e4e43f27475763f1d15b6d7c67

NikhilNarayana committed:
fix(build-linux): delete Sys folder before copy
e444dc8

NikhilNarayana committed:
chore: prevent generating playback data with ifdef
c2d676c

NikhilNarayana committed:
Remove updater
baca98a

NikhilNarayana committed:
add desync replay reporting
64feb0f

NikhilNarayana committed:
Load scenes via MxScn.dat
d275af7

NikhilNarayana committed:
feat: support ordered injection list traversal
1852ee5

NikhilNarayana committed:
fix: nametag display in old replays
86c3e16

r2dliu committed:
pull in 03c57192491178b329bda001c9fb8d8ba8a28c59

r2dliu committed:
fix warning

r2dliu committed:
pull in 8c627231616315db71709b13a38246a8bd7f183d

r2dliu committed:
pull in 686155bc0a5026113c02b3287e621dc11eb57353

r2dliu committed:
pull in bf0c0764a2905f91bb69c8aa273677b012fa89f0

r2dliu committed:
pull 5287899c4d374b38a9ae870db33615b53f29a61e

r2dliu committed:
even with f68f6f6cbf56b3e4e43f27475763f1d15b6d7c67. playback works, netplay is blank???

NikhilNarayana committed:
fix black screen with latest bootloader

NikhilNarayana committed:
default to netplay gecko codes

NikhilNarayana committed:
fix matchmaking

NikhilNarayana committed:
fix iso hashing and some cleanup

NikhilNarayana committed:
working build on linux

NikhilNarayana committed:
ci(macOS): remove libav

NikhilNarayana committed:
ci(linux): add libfuse2 to apt install

NikhilNarayana committed:
disable skipping frames after frame 120

f96178b47f3531e30ddbaaf68721db4586a09f0c to b1cea3a9570b0ee342d79200ec26c43b9def197b

NikhilNarayana committed:
mark replays as played on mainline

NikhilNarayana committed:
ci: use v3 actions

NikhilNarayana committed:
bring in rust stuff, music works. need to test osd and loggers a bit more

the rust stuff will get submoduled at some point

NikhilNarayana committed:
Determine the actual used local address with ENET socket (e39fe60)

NikhilNarayana committed:
pull in rust updates

NikhilNarayana committed:
Rust Game Reporter (deaddda)

NikhilNarayana committed:
actually commit corrosion?

NikhilNarayana committed:
don't strip libs in appimage

NikhilNarayana committed:
fix linking in linux

NikhilNarayana committed:
copy rust dlls to binary folder on windows

NikhilNarayana committed:
add note about rust in README

NikhilNarayana committed:
fix direct codes

NikhilNarayana committed:
rename exe to Slippi_Dolphin

NikhilNarayana committed:
fix usage of WriteStringToFile

file name comes first now

NikhilNarayana committed:
clang-format

NikhilNarayana committed:
delete SlippiRustExtensions locally

NikhilNarayana committed:
use SlippiRustExtensions submodule

NikhilNarayana committed:
bump version

NikhilNarayana committed:
MONTH_FOLDERS -> MONTHLY_FOLDERS

NikhilNarayana committed:
hook up most slippi pane settings

NikhilNarayana committed:
fix enabling rust logs on boot

NikhilNarayana committed:
clang-format

NikhilNarayana committed:
redo music volume control and refactor some stuff

rapito committed:
Fix macOS compilation errors (#81)

* Move #endif to proper location on UICommon.cpp preventing compilation because of syntax error

* Add missing include for unordered map and missing CodeTo implementation from Ishiiruka
+ Adjust code to use string_view

* Trick compiler to think var is being used on Semver200_parser.cpp to prevent compilation errors on macos

NikhilNarayana committed:
fix macOS CI

NikhilNarayana committed:
ci: specify dolphin-emu target

NikhilNarayana committed:
music controlled by asm and jukebox volume slider

NikhilNarayana committed:
chore: default cheats on

NikhilNarayana committed:
add windows CI builds (#82)

NikhilNarayana committed:
fix spectate functionality

NikhilNarayana committed:
ci: cleanup

NikhilNarayana committed:
snake_case_a_lot

NikhilNarayana committed:
needs_more_snake_case

NikhilNarayana committed:
bring exi slippi up to date

add desync recovery bits and some other clean up

NikhilNarayana committed:
finished read through of SlippiNetplay.cpp

NikhilNarayana committed:
finished read through of SlippiMatchmaking.cpp

NikhilNarayana committed:
finished read through of SlippiGameFileLoader.cpp

NikhilNarayana committed:
finished read through of SlippiExiTypes.cpp

NikhilNarayana committed:
finished read through of SlippiPad.cpp

NikhilNarayana committed:
finished read through of SlippiReplayComm.cpp

NikhilNarayana committed:
finished read through of SlippiSavestate.cpp

NikhilNarayana committed:
fix gameloader

NikhilNarayana committed:
actually scale to 73:60

NikhilNarayana committed:
remove unnecessary logger
